sabnzbd/downloader.py

```diff
@@ -912,6 +942,19 @@ def stop(self):
         self.shutdown = True
         notifier.send_notification("SABnzbd", T('Shutting down'), 'startup')
 
+    def purge_article_queue(self):
+        for server in self.servers:
+            if server.article_queue:
+                for article in server.article_queue:
+                    article.fetcher = None
+                    article.reset_try_list()
+                    article.nzf.reset_try_list()
+                    article.nzf.nzo.reset_try_list()
+                    article.tries -= 1
+                server.article_queue = []
+                logging.debug('article_queue wiped for %s', server.id)
```

Comment:
This function is the part I'm most in doubt about. I'm not sure if it is sufficient to avoid missing articles after shutdown, server restart or similar in all cases, or if it should be called more or different places than it currently is.